test/scale-out/cloud_scale_out_basic_auth_tls.bats

```diff
@@ -0,0 +1,159 @@
+
+    local zot_log_file="${BATS_FILE_TMPDIR}/zot-${zot_server_address}-${zot_server_port}.log"
+    echo "LOG: ${zot_log_file}" >&3
+
+    cat > ${zot_config_file}<<EOF
+{
+    "distSpecVersion": "1.1.0",
+    "storage": {
+        "rootDirectory": "${zot_root_dir}",
+        "dedupe": false,
+        "remoteCache": true,
+        "storageDriver": {
+            "name": "s3",
+            "rootdirectory": "/zot",
+            "region": "us-east-2",
+            "regionendpoint": "localhost:4566",
+            "bucket": "zot-storage-test",
+            "secure": false,
+            "skipverify": false
+        },
+        "cacheDriver": {
+            "name": "dynamodb",
+            "endpoint": "http://localhost:4566",
+            "region": "us-east-2",
+            "cacheTablename": "BlobTable",
+            "repoMetaTablename": "RepoMetadataTable",
+            "imageMetaTablename": "ImageMetaTable",
+            "repoBlobsInfoTablename": "RepoBlobsInfoTable",
+            "userDataTablename": "UserDataTable",
+            "apiKeyTablename":"ApiKeyTable",
+            "versionTablename": "Version"
+        }
+    },
+    "http": {
+        "address": "${zot_server_address}",
+        "port": "${zot_server_port}",
+        "realm": "zot",
+        "auth": {
+            "htpasswd": {
+                "path": "${ZOT_HTPASSWD_PATH}"
+            }
+        },
+        "tls": {
+            "cert": "${ROOT_DIR}/test/data/server.cert",
+            "key": "${ROOT_DIR}/test/data/server.key",
+            "cacert": "${ROOT_DIR}/test/data/ca.crt"
+        }
+    },
+    "cluster": {
+      "members": [],
+      "hashKey": "loremipsumdolors"
+    },
+    "log": {
+        "level": "debug",
+        "output": "${zot_log_file}"
+    }
+}
+EOF
+    update_zot_cluster_member_list_in_config_file ${zot_config_file} ${ZOT_CLUSTER_MEMBERS_PATCH_FILE}
+    zot_serve ${zot_config_file}
+    wait_zot_reachable ${zot_server_port}
+}
+
+# Setup function for single zot instance
+function setup() {
+    # Verify prerequisites are available
+    if ! $(verify_prerequisites); then
+        exit 1
+    fi
+
+    setup_cloud_services
+    setup_local_htpasswd
+
+    generate_zot_cluster_member_list ${NUM_ZOT_INSTANCES} ${ZOT_CLUSTER_MEMBERS_PATCH_FILE}
+
+    for ((i=0;i<${NUM_ZOT_INSTANCES};i++)); do
+        launch_zot_server 127.0.0.1 $(( 10000 + $i ))
+    done
+
+    # list all zot processes that were started
+    ps -ef | grep ".*zot.*serve.*" | grep -v grep >&3
+
+    setup_haproxy
+
+    # list haproxy processes that were started
+    ps -ef | grep "haproxy" | grep -v grep >&3
+}
+
+function teardown() {
+    local zot_root_dir=${BATS_FILE_TMPDIR}/zot
+    haproxy_stop_all
+    zot_stop_all
+    rm -rf ${zot_root_dir}
+    teardown_cloud_services
+}
+
+@test "Check for successful zb run on haproxy frontend" {
+    zb_run "cloud-scale-out-basic-auth-tls-bats" "https://127.0.0.1:8000" 1 1 "${ZOT_AUTH_USER}:${ZOT_AUTH_PASS}"
```

Comment:
zb add more clients and requests and also use a cidr
  -s, --src-cidr string        Use specified cidr to obtain ips to make requests from, src-ips and src-cidr are mutually exclusive
  -i, --src-ips string         Use colon-separated ips to make requests from, src-ips and src-cidr are mutually exclusive
  
  so that multiple client src IPs can be used.

Reply:
Same everywhere else.

Reply:
Added --src-cidr